brave/brave-core
# Deprecate CONTENT_SETTING_ASK for autoplay

Resolves https://github.com/brave/brave-browser/issues/13046

This PR
1. Migrates CONTENT_SETTING_ASK to current default
2. Discards ASK related codes which is no longer used

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.

## Test Plan:
1. Open 1.18.68(or any version doesn't contain https://github.com/brave/brave-core/pull/7197)  Brave on fresh profile 
2. Visit brave://settings/content/autoplay
3. Toggle the label from "Allowed"=>"Blocked" and then "Blocked"=>"Allowed"
4. Visit https://www.youtube.com/, there will be a autoplay prompt
5. Exit Brave and open `Preference` under profile, there is a `"default_content_setting_values": {"autoplay":3}` which is the ask value we want to get rid of
6. launch Brave with this fix
8. Visit brave://settings/content/autoplay, it should display "Allowed"
9. Visit https://www.youtube.com/, there should be no prompt.
10.  Exit Brave and open `Preference` under profile, there should not be any autoplay in `default_content_setting_values` because it is reset to default which is allow
